Moonscraper Chart Editor/Assets/Scripts/Game/NoteVisuals/NoteVisualsManager.cs

```diff
@@ -4,46 +4,107 @@
 using UnityEngine;
 using MoonscraperChartEditor.Song;
 
-public class NoteVisualsManager : MonoBehaviour {
+public class NoteVisualsManager : MonoBehaviour
+{
     public NoteController nCon;
     public TMPro.TextMeshPro text;
     [SerializeField]
     protected bool isTool = false;
     protected Renderer noteRenderer;
 
     [HideInInspector]
-    public Note.NoteType noteType = Note.NoteType.Strum;
+    public VisualNoteType noteType = VisualNoteType.Strum;
     [HideInInspector]
     public Note.SpecialType specialType = Note.SpecialType.None;
 
     Note prevNote;
 
     // Use this for initialization
-    protected virtual void Awake () {
+    protected virtual void Awake()
+    {
         noteRenderer = GetComponent<Renderer>();
+        ChartEditor.Instance.events.kickNoteRecolorToggledEvent.Register(UpdateVisualsCallback);
     }
 
     void LateUpdate()
-    {        
+    {
         Animate();
     }
 
-    public static Note.NoteType GetVisualNoteType(Note note)
+    private void UpdateVisualsCallback()
     {
-        Note.NoteType noteType = note.type;
+        if (nCon.note != null)
+            UpdateVisuals();
+    }
```

Comment:
Remove this, unnecessary as per previous comment. Set the object as dirty to cause a refresh of the visuals.